src/compiler/scanner.ts

```diff
@@ -1457,6 +1517,33 @@ module ts {
             pos = tokenPos;
             return token = scanTemplateAndSetTokenValue();
         }
+        
+        function reScanJsxIdentifier(): SyntaxKind {
+            if (token === SyntaxKind.Identifier) {
+                while (pos < end) {
+                    let ch = text.charCodeAt(pos);
+                    if (ch === CharacterCodes.minus || isIdentifierPart(ch)) {
+                        tokenValue += String.fromCharCode(ch);
+                        pos++;
+                    } 
+                    else  if (ch === CharacterCodes.backslash) {

```

Comment:
Need to bring in @fdecampedron on this one -- I don't know what the backslash means here either.